crates/ruff_linter/src/rules/ruff/rules/helpers.rs

```diff
@@ -84,3 +87,72 @@ pub(super) fn is_descriptor_class(func: &Expr, semantic: &SemanticModel) -> bool
         })
     })
 }
+
+// fast check to disqualify any string literal that doesn't
+// have opening and closing brackets
+#[inline]
+fn has_brackets(possible_fstring: &str) -> bool {
+    let mut opening = false;
+    let mut closing = false;
+    for c in possible_fstring.chars() {
+        match c {
+            '{' => opening = true,
+            '}' => closing = true,
+            _ => {}
+        }
+    }
+    opening && closing
+}
+
+/// Returns `true` if `source` is valid f-string syntax with qualified, bound variables.
+/// `kwargs` should be the keyword arguments that were passed to function if the string literal is also
+/// being passed to the same function.
+/// If a identifier from `kwargs` is used in `source`'s formatting, this will return `false`,
+/// since it's possible the function could be formatting the literal in question.
+pub(super) fn should_be_fstring(
+    source: &str,
+    kwargs: Option<&Vec<Keyword>>,
```

Comment:
I feel like I instinctively tend towards `Option<&[Keyword]>` for something like this. What are your preferences / tendencies?

Reply:
I prefer `Option<&[Keyword]>` too. This was just an oversight, my bad.